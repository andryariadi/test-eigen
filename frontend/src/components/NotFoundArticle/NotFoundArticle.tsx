const NotFoundArticle = () => {
  return (
    <div data-testid="not-found-article" className="flex items-center justify-center pb-10">
      <span className="text-xl text-center font-semibold text-slate-500">No articles found</span>;
    </div>
  );
};

export default NotFoundArticle;
