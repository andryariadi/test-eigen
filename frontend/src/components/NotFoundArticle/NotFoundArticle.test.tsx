import { render, screen } from "@testing-library/react";
import NotFoundArticle from "./NotFoundArticle";

describe.skip("NotFoundArticle Component", () => {
  it('renders correctly with the "No articles found" message', () => {
    render(<NotFoundArticle />);

    // Memeriksa elemen utama di-render
    const container = screen.getByTestId("not-found-article");

    expect(container).toBeInTheDocument();
    expect(container).toHaveClass("flex", "items-center", "justify-center", "pb-10");

    // Memeriksa teks pesan
    const message = screen.getByText("No articles found");

    expect(message).toBeInTheDocument();
    expect(message).toHaveClass("text-xl", "text-center", "font-semibold", "text-slate-500");
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<NotFoundArticle />);
    expect(asFragment()).toMatchSnapshot();
  });
});
