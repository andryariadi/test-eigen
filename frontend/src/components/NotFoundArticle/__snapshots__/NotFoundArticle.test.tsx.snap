// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`NotFoundArticle Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-center pb-10"
    data-testid="not-found-article"
  >
    <span
      class="text-xl text-center font-semibold text-slate-500"
    >
      No articles found
    </span>
    ;
  </div>
</DocumentFragment>
`;
